opentelemetry-api/tests/trace/test_status.py

```diff
@@ -0,0 +1,35 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+from opentelemetry.trace.status import Status, StatusCanonicalCode
+
+
+class TestStatus(unittest.TestCase):
+    def test_constructor(self):
+        status = Status()
+        self.assertEqual(status.canonical_code, StatusCanonicalCode.OK)
+        self.assertEqual(status.description, None)
+
+        status = Status(StatusCanonicalCode.UNAVAILABLE, "unavailable")
+        self.assertEqual(
+            status.canonical_code, StatusCanonicalCode.UNAVAILABLE
+        )
+        self.assertEqual(status.description, "unavailable")
+
+    def test_invalid_description(self):
+        status = Status(description={"test": "val"})
```

Comment:
Include a test checking that a warning is printed?

Reply:
Added.